### PR TITLE
Debian installer script, armui changes, AppRise notifications

### PR DESCRIPTION
**Pull request by 1337-server**
Debian installer scripts
Armui changes - more detailed server stats, Icons in the history page to make things clearer, etc
AppRise notifications
A number of small bug fixes
Added ability to disabled ripping the same dvd more than once
Added moving of identified dvd files to the media folder

This should satisfy #368

### CODE REVIEW

**Comment by 1337-server** [comment]
Has anyone has a chance to review any of these changes ?

**Comment by flammableliquids** [comment]
> 
> 
> Has anyone has a chance to review any of these changes ?

I will be trawling through this between now and the Christmas break

**Comment by 1337-server** [comment]
> > Has anyone has a chance to review any of these changes ?
> 
> I will be trawling through this between now and the Christmas break
Thats great!

I will say that i have changed identify.py locally to use your version of  identify_dvd() and callwebservice() with some small changes . But i havent updated the pull request with these changes. So i would ignore the changes to both callwebservice() and identify_dvd() Ill hopefully get around to updating the identify.py this week. So it can be much more easily merged.

**Comment by 1337-server** [comment]
I have added a fix to update the MAINFEATURE in the database. I have also added the changes from pull #318 as these were very simple to add to the UI

**Review by flammableliquids — CHANGES_REQUESTED** [review]
This is pointing to your own personal repo for the installation script. Any chance we could move the installation script into the main?

**Comment by 1337-server** [comment]
Fixed!

File: arm/ripper/main.py
Comment (flammableliquids) [inline arm/ripper/main.py:188]: You do a check for duplicates below, but the check of duplicate is based on if the folder exists in the target destination.
Something I had on my future plans for this project was to have the scripts check to historical jobs to see if the CRC of the disc in play matches anything that already exists. If the CRC matches, use the title and year from that entry. 
(I wanted this functionality to reduce wasted time when I am repeatedly ripping the same disc while tweeking the HandBrake parameters (Before you start, I am aware that there may be problems with multiple matches).

It might be a more robust solution to check the jobs table for successful jobs with a matching CRC to qualify something as a dupe. From what I have seen a number of people use their destination directory as more of a 'staging ground' before it hit's their plex or emby server.

File: arm/ui/routes.py
Comment (flammableliquids) [inline arm/ui/routes.py:115]: I'm not sure if this is nit-picking, but could we move the 'flash' to below the db.session.refresh's, and have the Flash use the actual values in the job and config?

File: arm/ui/routes.py
Comment (flammableliquids) [inline arm/ui/routes.py:259]: I had thought about something like this, but hesitated as I felt if I did CPU, I should probably do GPU utilization (nvidia-smi -l )

File: arm/ui/templates/login.html
Comment (flammableliquids) [inline arm/ui/templates/login.html:20]: I can't seem to find where the username and password is set.

File: scripts/debian-setup.sh
Comment (flammableliquids) [inline scripts/debian-setup.sh:109]: I'm a little confused. Why have you specified Debian? The project was specific in utilizing Ubuntu. If this script is designed to only work for Debian, wouldn't it be better to modify it to work on Ubuntu and have a second setup for Debian?
Btw, LOVE the setup of the WebUI as a service.

File: setup/.abcde.conf
Comment (flammableliquids) [inline setup/.abcde.conf:99]: I'm excited to test this out. I had quite a lot of problems with disk recognition. And I was thinking that we might want to update the documentation to include finding the offset of your specific drive.

File: scripts/debian-setup.sh
Comment (flammableliquids) [inline scripts/debian-setup.sh:76]: This might be something that should eventually become parameter based, but I think we should leave that until someone complains.

File: scripts/debian-setup.sh
Comment (1337-server) [reply scripts/debian-setup.sh:109]: I personally use debian. However i did make a install script for both (i didnt include the Ubuntu installer because i wasnt sure if it would be used). 
The main reason i created the setup script was for myself to automate the install process. Simply following the commands on the front page wouldnt get ARM installed.
You can see the instruction for ubuntu and the ubuntu install script [here](https://github.com/1337-server/automatic-ripping-machine/wiki/Setting-up-ARM-script-(Ubuntu)).

File: arm/ui/templates/login.html
Comment (1337-server) [reply arm/ui/templates/login.html:20]: I appologize, these are never set in this PR. I added it for later use. I have since implemented a login page & security , but didnt want to increase the size of this already large PR.

File: arm/ripper/main.py
Comment (1337-server) [reply arm/ripper/main.py:188]: The dupe check could be much better. The only reason i included this is that, it solves a number of issues.

- It solves the "not moving" issue (Bug where ARM will not move correctly identied movies)
- it solves the "duplicate jobs" issue  where jobs trigger more than once. (Bug/ possibly faulty drives issue)

But i agree that a CRC lookup with successful jobs would be a much better way of checking for if the disc is a dupe or not.

**Comment by flammableliquids** [comment]
The code looks good. I am going to do a fresh re-install of my whole system over the weekend to test out the instructions, readme, installation scripts, gui changes, and the added web-ui authentication.

File: arm/ui/templates/login.html
Comment (flammableliquids) [reply arm/ui/templates/login.html:20]: Ok, I kept trying to find the location where the cookies and password authent was forced.

File: setup/.abcde.conf
Comment (1337-server) [reply setup/.abcde.conf:99]: One of the main problems for ARM with Music CD's currently, the master branch is set to use CDDB but they have long since closed their doors. They closed around the same time Microsoft shut down their CRC metadata service.

This updates the config to use MusicBrainz and also adds the Musicbrains module to ARM allowing the Disc to be correctly identified in both ARM and ARMui.
Musicbrains do have a searchable database, via API that could allow users to search for disc, but because musicbrainz use a specific way of hashing each disc i didnt find it ever necessary to manually search.

File: scripts/debian-setup.sh
Comment (1337-server) [reply scripts/debian-setup.sh:76]: This is possible in bash, I wasnt sure what the best default action was so i left it as simple as possible.

**Comment by 1337-server** [comment]
> The code looks good. I am going to do a fresh re-install of my whole system over the weekend to test out the instructions, readme, installation scripts, gui changes, and the added web-ui authentication.

That sounds great. Just a small caution. If you use my own repo to pull the changes you need to specifically checkout v2_master branch. My default branch is v2.4 and it has a lot of changes from this PR.

File: arm/ui/routes.py
Comment (1337-server) [reply arm/ui/routes.py]: Ive commited these changes.

File: arm/ui/routes.py
Comment (1337-server) [reply arm/ui/routes.py:258]: The only reason i included the cpu is because it was easy to show what cpu the user is using no matter if they are using Intel or AMD.
This also helped for me personally, to make it clear if i was on my main servers page or my testing rig.

The only problem i see with adding GPU usage is that AMD GPU's and Intel APU's would not be supported. We have no idea on the percentages these are distributed in users. So its hard to know what percentage of users these changes would be useful too.

It is possible to very easily show CPU usage thanks to the already loaded psutil module. Its something i have already added in my own repo.

File: arm/ripper/main.py
Comment (1337-server) [reply arm/ripper/main.py:188]: I managed to make a much better check for dupe entries using the database.

For testing this make things sooo much easier. Ive written it so that even if dupe rips are disabled it will allow through rips until a successful rip is added to the database. Then it fully disables re-rips.
It also takes the confirmed details from the previous rips. 

So if you have a disc that doesnt get ident when using OMdb and you search/add details, all future dupes will try to use the details you have set.

I still need to test it more in VMs/main server. However it works great on my test server. Great idea!

**Comment by cmbernard333** [comment]
I checked this branch out recently and was unable to get it to run properly. It seems like its a little out of date with `v2_master` as well. Is there any movement on this PR?

**Comment by 1337-server** [comment]
Do you mind sharing more details of what you couldn't get working ? 

It's been quite a while since I last tested this PR, however I was sure that it was still fully functioning. 

This PR is still waiting for approval from flammingliquid. I made all the requested changes, but no word back from them yet.

**Comment by cmbernard333** [comment]
I did these steps:
```
$ git remote add 1337-server <github_url>
$ git checkout -b 1337-server/v2_master 1337-server/v2_master
$ sudo udevadm control --reload-rules; sudo udevadm monitor # watching events here
$ eject  # wait 30 seconds or so
$ eject -t 
$ ps -ef | grep py # looking for ARM here
```
Process shows up but it either creates a None_*.log in /home/arm/logs or doesn't do anything at all.

**Comment by 1337-server** [comment]
> 
> 
> I did these steps:
> 
> ```
> $ git remote add 1337-server <github_url>
> $ git checkout -b 1337-server/v2_master 1337-server/v2_master
> $ sudo udevadm control --reload-rules; sudo udevadm monitor # watching events here
> $ eject  # wait 30 seconds or so
> $ eject -t 
> $ ps -ef | grep py # looking for ARM here
> ```
> 
> Process shows up but it either creates a None_*.log in /home/arm/logs or doesn't do anything at all.
> 
> EDIT: I watched `sudo iotop` as well to see if `makemkv` was running and it never showed up.

Did you rerun `sudo pip3 install -r requirements.txt` ?

The pull has new requirements that done come installed with previous versions of ARM

**Comment by cmbernard333** [comment]
I'll try that out. This is a much needed change because I run into multiple instances of arm daily and it doesn't stop till I kill all instances and reset udev.

**Comment by cmbernard333** [comment]
Things seem to be working now. I have been testing bluray discs for the past day or so. What can we do to get this merged to master?

**Comment by 1337-server** [comment]
> Things seem to be working now. I have been testing bluray discs for the past day or so. What can we do to get this merged to master?

Glad it's working. 

Still waiting for flamingliquid to approve the changes so they can be merged.

**Comment by flammableliquids** [comment]
The instructions for v2.2 are a little lacking. But it seems to work ok.
I'm going to wipe and restore again to test.

**Pull request merged**
